cpp/modmesh/buffer/ConcreteBuffer.cpp

```diff
@@ -0,0 +1,107 @@
+/*
```

Comment:
not thing new, only copy and paste

Reply:
This file is too short to make distinct implementation file useful.  It does not make code easier to read.  In contrary, I think it makes it harder to read the code because function bodies are moved away from the header file.

And I think the code split in the new file is too short to significantly shorten the compilation time.  It is so short that measuring it rigorously probably does not make sense.